### PR TITLE
[toolsrc] Optionally allow vcpkg to clean after each build

### PR DESCRIPTION
**Pull request by seanwarren**
vcpkg can consume very significant amounts of space when building multiple ports. This can cause problems operating in resource constrained environments, e.g. a CI pipeline. 

This PR adds switch `--clean-after-build`. When used, vcpkg will delete the port `packages`, `buildtrees` folder and any non-tool downloads after each port build completes. This minimises the footprint of the overall build.

### CODE REVIEW

File: toolsrc/src/vcpkg/install.cpp [hunk 1]
```diff
@@ -463,14 +473,16 @@ namespace vcpkg::Install
...
 
-    static constexpr std::array<CommandSwitch, 6> INSTALL_SWITCHES = {{
+    static constexpr std::array<CommandSwitch, 7> INSTALL_SWITCHES = {{
         {OPTION_DRY_RUN, "Do not actually build or install"},
         {OPTION_USE_HEAD_VERSION, "Install the libraries on the command line using the latest upstream sources"},
         {OPTION_NO_DOWNLOADS, "Do not download new sources"},
         {OPTION_RECURSE, "Allow removal of packages as part of installation"},
         {OPTION_KEEP_GOING, "Continue installing packages on failure"},
         {OPTION_USE_ARIA2, "Use aria2 to perform download tasks"},
+        {OPTION_CLEAN_AFTER_BUILD, "Clean buildtrees and packages after building each package"},
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/install.cpp]: Should probably indicate that it also purges archive downloads as well.

Comment (seanwarren) [reply toolsrc/src/vcpkg/install.cpp]: Thanks, I've now clarified this point

File: toolsrc/src/vcpkg/commands.ci.cpp [hunk 2]
```diff
@@ -229,6 +229,7 @@ namespace vcpkg::Commands::CI
             Build::AllowDownloads::YES,
             Build::CleanBuildtrees::YES,
             Build::CleanPackages::YES,
+            Build::CleanDownloads::YES,
```
Comment (vicroms) [inline toolsrc/src/vcpkg/commands.ci.cpp]: @Rastaban 

Just to make sure, are you OK with this being the default for `vcpkg ci`?

Comment (seanwarren) [reply toolsrc/src/vcpkg/commands.ci.cpp]: Ah sorry I didn't intend to modify the vcpkg `ci` defaults, shall I revert to not cleaning downloads? This probably isn't desirable if you have persistent build servers

Comment (Rastaban) [reply toolsrc/src/vcpkg/commands.ci.cpp]: The current state of our system will delete downloads that has not been accessed in the past week, so we keep them around for awhile.  We should probably keep that behavior for now.

Comment (Rastaban) [reply toolsrc/src/vcpkg/commands.ci.cpp]: I am interested in moving to build servers that are not persistent sometime in the future though, at that point we will may want to enable it in the CI system.

Comment (seanwarren) [reply toolsrc/src/vcpkg/commands.ci.cpp]: OK I've set `Build::CleanDownloads::NO` in `command.ci.cpp`

**Comment by vicroms** [comment]
@seanwarren thanks for your contribution!

**Pull request merged**